core-plugins/src/main/java/co/cask/hydrator/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -0,0 +1,92 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.joiner;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.plugin.PluginConfig;
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.annotation.Nullable;
+
+/**
+ * Config for join plugin.
+ */
+public class JoinerConfig extends PluginConfig {
+  private static final String NUM_PARTITIONS_DESC = "Number of partitions to use when joining. " +
+    "If not specified, the execution framework will decide how many to use.";
+  private static final String JOIN_KEY_DESC = "List of join keys to perform join operation. The list is " +
+    "separated by join condition. Join key from each input stage will be prefixed with '<stageName>.' And the " +
+    "relation among join keys from different inputs is represented by '='. For example: customers.customer_id=items" +
+    ".c_id&customers.customer_name=items.c_name will mean the join key is a composite key of customer id and customer" +
+    " name from customers and items input stages and join will be performed on equality of the join keys.";
+  private static final String SELECTED_FIELDS = "Map which represents fields to be selected and " +
+    "renamed in join output from each input stage. JSON string of map of map with stage name being key and value " +
+    "being map of old field name to alias. There should be at least one selected field. There must not be any " +
+    "duplicate fields in output.";
+  private static final String REQUIRED_INPUTS_DESC = "Comma-separated list of stages for which required input stages." +
+    " Required input stages decide the type of the join. If all the input stages are present in required inputs, " +
+    "inner join will be performed. Otherwise, outer join will be performed considering non-required inputs as " +
+    "optional.";
+
+
+  @Nullable
+  @Description(NUM_PARTITIONS_DESC)
+  protected Integer numPartitions;
+
+  @Description(JOIN_KEY_DESC)
+  protected String joinKeys;
+
+  @Nullable
+  @Description(SELECTED_FIELDS)
+  protected String selectedFields;
+
+  @Nullable
+  @Description(REQUIRED_INPUTS_DESC)
+  protected String requiredInputs;
+
+  public JoinerConfig() {
+    this.joinKeys = "";
+    this.selectedFields = "";
+    this.requiredInputs = "";
+  }
+
+  @VisibleForTesting
+  JoinerConfig(String joinKeys, String selectedFields, String requiredInputs) {
+    this.joinKeys = joinKeys;
+    this.selectedFields = selectedFields;
+    this.requiredInputs = requiredInputs;
+  }
+
+  @Nullable
+  public Integer getNumPartitions() {
+    return numPartitions;
+  }
+
+  @Nullable
+  public String getSelectedFields() {
+    return selectedFields;
+  }
+
+  public String getJoinKeys() {
+    return joinKeys;
+  }
+
+  @Nullable
```

Comment:
right but the no-args constructor sets it to an empty string, which means this can never return null. I think you want to set selectedFields and requiredInputs to null in the no-args constructor.